quicknote.rb

```diff
@@ -3,29 +3,48 @@
 require 'sinatra'
 require 'sinatra/reloader'
 require 'json'
+require 'pg'
 
-FILE_PATH = 'data/quicknote.json'
+conn = PG.connect(
+  dbname: ENV['DB_NAME'],
+  user: ENV['DB_USER'],
+  password: ENV['DB_PASSWORD'],
+  host: ENV['DB_HOST'], client_encoding: 'UTF-8'
+)
 
 helpers do
   def h(text)
     Rack::Utils.escape_html(text)
   end
 end
 
-def load_notes(file_path)
-  File.open(file_path) { |f| JSON.parse(f.read) }
+def read_notes(conn)
+  data = {}
+  result = conn.exec('SELECT * FROM note')
+  result.each do |row|
+    data[row['id']] = { 'title' => row['title'], 'content' => row['content'] }
+  end
+  data
```

Comment:
豆知識程度で、強いて修正する必要はありませんが、 `each_with_object` で少し簡単に書けると思います

```suggestion
  result.each_with_object({}) do |row, memos|
    memos[row['id']] = { 'title' => row['title'], 'content' => row['content'] }
  end
```

https://docs.ruby-lang.org/ja/latest/method/Enumerable/i/each_with_object.html